app/database/manager/index.ts

```diff
@@ -103,6 +107,50 @@ class DatabaseManager {
       this.androidFilesDirectory = Platform.OS === 'android' ? FileSystem.documentDirectory : null;
   }
 
+  /**
+     * getDatabaseConnection: Given a server url (serverUrl) and a flag (setAsActiveDatabase), this method will attempt
+     * to retrieve an existing database connection previously created for that url.
+     * If not found, it will create a new connection and register it in the DEFAULT_DATABASE. In this case, it will also
+     * use the provided connectionName as display_name for this server
+     * @param {GetDatabaseConnectionArgs} getDatabaseConnection
+     * @param {string} getDatabaseConnection.connectionName
+     * @param {string} getDatabaseConnection.serverUrl
+     * @param {boolean} getDatabaseConnection.setAsActiveDatabase
+     * @returns {Promise<DatabaseInstance>}
+     */
+  getDatabaseConnection = async ({serverUrl, setAsActiveDatabase, connectionName}: GetDatabaseConnectionArgs) => {
+      // We potentially already have this server registered; so we'll try to retrieve it if it is present under DEFAULT_DATABASE/GLOBAL entity
+      const existingServers = await this.retrieveDatabaseInstances([serverUrl]) as RetrievedDatabase[];
+
+      // Since we only passed one serverUrl, we'll expect only one value in the array
+      const serverDatabase = existingServers?.[0];
+
+      let connection: DatabaseInstance;
+
+      if (serverDatabase) {
+          // This serverUrl has previously been registered on the app
+          connection = serverDatabase.dbInstance;
```

Comment:
We are using different terms but each refer to the same thing.   A databaseInstance is of type Database or undefined in case WDB was not able to create a connection to the *.db file.  A connection is a valid pointer to a database file.


The `retrieveDatabaseInstances` method returns `return {dbInstance, displayName, url};` if a matching url has been found under the `Servers` entity in the `default` database.

Imagine we have `multi-server support`.  The user opens the side-bar to perform a `server switch`.  Instead of returning an array of object, we were returning only the connections.  I find it easier to identify connections using their url/displayName rather than retrieving it from inside the `dbInstance` object.